version: '3'

services:

  redis:
    image: redis
    ports:
      - '6379:6379'
    healthcheck:
      test: [ "CMD-SHELL", "redis-cli ping" ]
      interval: 10s
      timeout: 5s
      retries: 5

  db:
    image: postgres
    environment:
      - POSTGRES_DB=atomic_habits
      - POSTGRES_USER=postgres
      - POSTGRES_PASSWORD=examplepassword
      - PGDATA=/var/lib/postgresql/data/pgdata
    volumes:
      - pg_data:/var/lib/postgresql/data/pgdata
    ports:
      - '5432:5432'
    healthcheck:
      test: [ "CMD-SHELL", "pg_isready -U postgres -d atomic_habits" ]
      interval: 10s
      timeout: 5s
      retries: 5

  app:
    build: .
    tty: true
    command: python manage.py migrate && python manage.py runserver 0.0.0.0:8000
    ports:
      - '8001:8000'
    volumes:
      - .:/code
    env_file: .env.sample
    depends_on:
      redis:
        condition: service_healthy
      db:
        condition: service_healthy

  celery:
    build: .
    env_file: .env.sample
    tty: true
    command: celery -A config worker -l INFO
    depends_on:
      - app
      - redis

  celery_beat:
    build: .
    env_file: .env.sample
    tty: true
    command: celery -A config beat -l INFO -S django
    depends_on:
      - app
      - redis

volumes:
  pg_data: